src/types/types.ts

```diff
@@ -724,6 +862,36 @@ export interface Filter {
 
 export interface FilterQuery extends Cursor, Limit {}
 
+interface SequenceRowsRetriveData extends Cursor, Limit {
+  /**
+   * Lowest row number included.
+   * @default 0
+   */
+  start?: number;
+  /**
+   * Get rows up to, but excluding, this row number.
+   * @default "no limit"
+   */
+  end?: number;
+  /**
+   * Columns to be included. Specified as list of column externalIds.
+   * In case this filter is not set, all available columns will be returned.
+   */
+  columns?: string[];
+}
+
+/**
+ * A request for datapoints stored
+ */
+export type SequenceRowsRetrieve = SequenceRowsRetriveData & IdEither;
```

Comment:
Same here. Interface that extends

Reply:
IdEither is a union type, when you do `SequenceRowsRetriveData & IdEither` it actually creates type, `(SequenceRowsRetriveData & InternalId) | (SequenceRowsRetriveData & ExternalId)`
How will you do it with extends?